packages/block-editor/src/components/list-view/use-list-view-images.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+function getImageUrl( block ) {
```

Comment:
Not for this PR, but it's be good to think through an API that goes beyond hard-coding core blocks and builds similarly to the custom label support. For example, it might be a side effect of attribute types, so that a Cover block can show an image if one is set as well.

Reply:
Oh, thanks for taking an early look! Yes, I've just hard-coded things in at the moment while I hack around, but it'd be preferable in the longer-term to have a consistent API for it.

I imagine we'd likely keep things fairly hard-coded for the first round, but thinking through an API could be handy so that custom blocks / plugins can have image previews, too.

Reply:
Indeed, one step at a time :)

Reply:
Maybe in addition to the `save` and `edit` functions there could be a `list` function or something that returns JSX that appears in the List View. This could replace `__experimentalLabel`.

```js
list( { attributes } ) {
	return (
		<Row>
			__( 'Image' )
			<ImagePreview attributes={ attributes } />
		</Row>
	);
}
```

Reply:
Hrm, interesting idea! One thing we'd probably want to be careful about is that it's quite a restricted area visually, so we might not want custom blocks to arbitrarily add elements, but rather provide data that the UI already knows how to display without anything overflowing or breaking out of the available space? There isn't very much real estate available, and the text truncation, etc for `__experimentalLabel` is handled automatically by the list view rather than individual blocks having to worry about it.

Reply:
It's a similar problem to `BlockToolbar`. In theory you can put anything inside a `<BlockControls>` and it will appear in the toolbar but we trust that extenders use the components we provide that are designed for this purpose e.g. `ToolbarButton`.

In fact the problems are so similar that maybe instead of a `list()` function we just have `ListViewControls` which is rendered by `edit()` 😀

Reply:
> In fact the problems are so similar that maybe instead of a list() function we just have ListViewControls which is rendered by edit() 😀

Yeah, something like that could work. Though I'd imagine that would likely be in addition to a label function? `__experimentalLabel` works for other contexts than just the list view, since it winds up being called internally by [`useBlockDisplayTitle`](https://github.com/WordPress/gutenberg/blob/9764831790021019633f4227be50cbdfa359aba4/packages/block-editor/src/components/block-title/use-block-display-title.js#L72).

In any case, definitely worth hacking around with in follow-ups to see if it's more ergonomic for the block code to decide what's rendered, or for the block code to provide data that the list view renders in an opinionated way.